server/controllers/request.js

```diff
@@ -77,3 +80,34 @@ exports.updateRequest = asyncHandler(async (req, res, next) => {
     throw new Error("Invalid data");
   }
 });
+
+// @route POST /request/pay
+// @desc pay for a request
+// @access Private
+exports.payRequest = asyncHandler(async (req, res, next) => {
+  const { id, amount, description } = req.body;
+  const userId = req.user.id;
+  const profile = await Profile.findOne({ user: userId});
+  const paymentMethod = await stripe.paymentMethods.retrieve(profile.paymentId);//get paymentMethod
+  
+  //pay the request
+  const payment = await stripe.paymentIntents.create({
+    amount:amount,
+    currency: 'CAD',
+    customer: paymentMethod.customer,
+    description: description,
+    payment_method: paymentMethod.id,
+    confirm: true,
+  });
+  //if successs, set the paid field to true 
+  if (payment) {
+    const request = await Request.findByIdAndUpdate(
+      { _id: id },
+      { $set: { paid: true } },
+      { new: true }
+    );
+   return res.status(201).json({ request });
+  }
+  res.status(400);
+  throw new Error("something went wrong");
```

Comment:
This error message is very generic and not informative at all, let's reword this to add a bit more context. 😄